state-chain/pallets/cf-ingress-egress/src/boost_pool.rs

```diff
@@ -404,6 +404,34 @@ where
 		booster_contributions.len()
 	}
 
+	// If some funds were mistakenly marked as lost, this function can be used to recover them.
+	// It reverses the effects of `process_deposit_as_lost` and then applies
+	// `process_deposit_as_finalised`.
+	pub fn process_deposit_as_recovered(
+		&mut self,
+		prewitnessed_deposit_id: PrewitnessedDepositId,
+		booster_contributions: BTreeMap<AccountId, OwedAmountScaled<C>>,
+		pending_withdrawals: BTreeSet<AccountId>,
+	) -> Vec<(AccountId, C::ChainAmount)> {
+		if self
+			.pending_boosts
+			.insert(prewitnessed_deposit_id, booster_contributions)
+			.is_some()
+		{
+			log::warn!("Can't recover a deposit if it already exists. Aborting.");
+			return Default::default();
+		}
+
+		for booster_id in pending_withdrawals {
```

Comment:
We need to be sure that `pending_withdrawals` only include accounts who contributed to the "lost" deposit (i.e. not just accounts who were withdrawing at the time, though it is possible that they are the same sets). Otherwise these pending_withdrawals entries will never be removed. I think we can check this by looking at `booster_contributions`.

However, I see that we are providing an empty set in the migration, so this won't matter in that case.